### PR TITLE
add some user inteface

### PR DESCRIPTION
**Pull request by YasiucheniaAndrey**

### CODE REVIEW

File: djangoProject/settings.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 DEBUG = True
 
 ALLOWED_HOSTS = []
+EMAIL_HOST = "localhost"
+EMAIL_PORT = 1025
```
Comment (revimi) [inline djangoProject/settings.py:30]: Is it used for sending emails?

File: djangoProject/settings.py [hunk 2]
```diff
@@ -94,20 +96,21 @@
...
+    # },
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
+    # },
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
+    # },
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
+    # },
```
Comment (revimi) [inline djangoProject/settings.py:110]: Please uncomment

File: mysite/urls.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from django.urls import path
+from . import views
+from django.conf.urls import include
+
+
+#app_name = 'category'
```
Comment (revimi) [inline mysite/urls.py:6]: Do we need this line?

File: mysite/urls.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+urlpatterns = [
+#    path('<int:pk>/', views.ProdeuctDetailView.as_view(), name='detail'),
```
Comment (revimi) [inline mysite/urls.py:8]:
```suggestion
#    path('<int:pk>/', views.ProductDetailView.as_view(), name='detail'),
```

File: mysite/urls.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    path("categories/", views.CategoryView.as_view(), name='index'),
+    path("dashboard/", views.dashboard, name="dashboard"),
+    path("register/", views.register, name="register"),
+    path("accounts/", include("django.contrib.auth.urls"))
```
Comment (revimi) [inline mysite/urls.py:12]: Why are the auth urls included twice?

Comment (revimi) [reply mysite/urls.py:12]: see [this line](https://github.com/YasiucheniaAndrey/djangoProject/pull/5/files?diff=split&w=0#diff-86bf0aeb5e4ec395f0997accb74f6afc7ce52964a229ef00b87644e5803ec6dcR25)

File: mysite/views.py [hunk 6]
```diff
@@ -1,3 +1,30 @@
...
+            user = form.save()
+            login(request, user)
+            return redirect(reverse("dashboard"))
+
+class CategoryView(generic.ListView):
+    template_name = 'mysite/categories.html'
+    context_object_name = 'categories_list'
+
+    def get_queryset(self):
+        return Category.objects.all()
```
Comment (revimi) [inline mysite/views.py:29]: Unrelated to this PR. Please move to another one

File: product/views.py [hunk 7]
```diff
@@ -5,21 +5,29 @@
...
 
 
+class ProductListViewByCategory(generic.ListView):
+    template_name = 'product/products_by_category.html'
+    model = Product
+
+    # def get_queryset(self, *args, **kwargs):
+    #     qs = super(ProductListViewByCategory, self).get_queryset()
+    #     qs = qs.order_by("-id")
+    #     return qs
```
Comment (revimi) [inline product/views.py:23]: Why is it commented out?

File: templates/mysite/categories.html [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    <ul>
+    {% for category in categories_list %}
+        <li><a href="/product/{{ category.name }}/">{{ category.name }}</a></li>
+    {% endfor %}
+    </ul>
+{% else %}
+    <p>No products are available.</p>
+{% endif %}
+
+
```
Comment (revimi) [inline templates/mysite/categories.html:12]: Not related to users - let's have a separate PR

**Review by revimi — COMMENTED** [review]
Ну вообще в целом неплохо

File: djangoProject/settings.py (hunk 1 above)
Comment (YasiucheniaAndrey) [reply djangoProject/settings.py:30]: it is for debug purpose, removed

File: djangoProject/settings.py (hunk 2 above)
Comment (YasiucheniaAndrey) [reply djangoProject/settings.py:110]: ok

File: mysite/urls.py (hunk 3 above)
Comment (YasiucheniaAndrey) [reply mysite/urls.py:6]: removed

File: product/views.py (hunk 7 above)
Comment (YasiucheniaAndrey) [reply product/views.py:23]: in progress, not working now

File: mysite/urls.py (hunk 4 above)
Comment (YasiucheniaAndrey) [reply mysite/urls.py:8]: ?

File: mysite/urls.py (hunk 5 above)
Comment (YasiucheniaAndrey) [reply mysite/urls.py:12]: removed in head urls

File: mysite/urls.py (hunk 4 above)
Comment (revimi) [reply mysite/urls.py:8]: опечатка

**Review by revimi — APPROVED** [review]

**Pull request merged**